### ISSUE TITLE
Inconsistent behavior related to VRRP packets drops

### ISSUE THREAD

**Issue by ooraini** [cilium/cilium#18347]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### What happened?

I'm running keepalived on two of my hosts and they are both reporting a timeout.
The VIP is 192.168.33.200. Running `ip ad | grep 200` on the two nodes:
```
node3 | CHANGED | rc=0 >>
    inet 192.168.33.200/32 scope global eth1
node2 | CHANGED | rc=0 >>
    inet 192.168.33.200/32 scope global eth1
```

Looking at `hubble observe -t drop` I see the following:
```
Dec 30 21:01:55.186: 192.168.33.20 <> 192.168.33.30 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:55.808: 192.168.33.30 <> 192.168.33.20 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:56.186: 192.168.33.20 <> 192.168.33.30 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:56.415: fe:54:00:7f:be:9e <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:56.472: fe:54:00:34:d6:c7 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:56.702: fe:54:00:e3:4a:21 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:56.808: 192.168.33.30 <> 192.168.33.20 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:56.829: fe:54:00:4b:40:45 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:56.861: fe:54:00:1e:82:f6 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:57.140: fe:54:00:2c:b7:f6 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:57.186: 192.168.33.20 <> 192.168.33.30 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:57.808: 192.168.33.30 <> 192.168.33.20 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:58.186: 192.168.33.20 <> 192.168.33.30 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:58.456: fe:54:00:34:d6:c7 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:58.686: fe:54:00:e3:4a:21 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
Dec 30 21:01:58.808: 192.168.33.30 <> 192.168.33.20 CT: Unknown L4 protocol DROPPED (IPv4)
Dec 30 21:01:59.124: fe:54:00:2c:b7:f6 <> 01:80:c2:00:00:00 Unsupported L2 protocol DROPPED (Ethernet)
```

JSON output:
```
{"time":"2021-12-30T21:06:17.103685380Z","verdict":"DROPPED","drop_reason":166,"ethernet":{"source":"fe:54:00:2c:b7:f6","destination":"01:80:c2:00:00:00"},"source":{"identity":1,"labels":["reserved:host"]},"destination":{"identity":2,"labels":["reserved:world"]},"Type":"L3_L4","node_name":"node2","event_type":{"type":1,"sub_type":166},"traffic_direction":"INGRESS","drop_reason_desc":"UNSUPPORTED_L2_PROTOCOL","Summary":"Ethernet"}
{"time":"2021-12-30T21:06:17.210109754Z","verdict":"DROPPED","drop_reason":137,"ethernet":{"source":"52:54:00:2c:b7:f6","destination":"52:54:00:4b:40:45"},"IP":{"source":"192.168.33.20","destination":"192.168.33.30","ipVersion":"IPv4"},"source":{"identity":1,"labels":["reserved:host"]},"destination":{"identity":6,"labels":["reserved:remote-node"]},"Type":"L3_L4","node_name":"node2","event_type":{"type":1,"sub_type":137},"traffic_direction":"INGRESS","drop_reason_desc":"CT_UNKNOWN_L4_PROTOCOL","Summary":"IPv4"}
```

The two nodes IPs are 192.168.33.20 and 192.168.33.30.

I've tested with both 1.11 and 1.10 with exactly identical machines(except for Cilium) and the issue only happens on 1.11

### Cilium Version

```
Defaulted container "cilium-agent" out of: cilium-agent, clean-cilium-state (init)
Client: 1.11.0 27e0848 2021-12-05T15:34:41-08:00 go version go1.17.3 linux/amd64
Daemon: 1.11.0 27e0848 2021-12-05T15:34:41-08:00 go version go1.17.3 linux/amd64
```

### Kernel Version

`Linux node1 4.18.0-348.7.1.el8_5.x86_64 #1 SMP Tue Dec 21 19:02:23 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`

### Kubernetes Version

```
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:16:20Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-07T18:09:57Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}

```

### Sysdump

[https://file.io/Aai7Vju4o9I3](https://file.io/Aai7Vju4o9I3)

### Anything else?

`cilium status --verbose`:
```
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.23 (v1.23.0) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Strict    [eth1 192.168.33.20 (Direct Routing)]
Host firewall:          Enabled   [eth1]
Cilium:                 Ok   1.11.0 (v1.11.0-27e0848)
NodeMonitor:            Listening for events on 2 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok   
IPAM:                   IPv4: 3/254 allocated from 172.16.1.0/24, 
Allocated addresses:
  172.16.1.134 (router)
  172.16.1.196 (ingress-nginx/ingress-nginx-controller-njn5d)
  172.16.1.26 (health)
BandwidthManager:       Disabled
Host Routing:           Legacy
Masquerading:           IPTables [IPv4: Enabled, IPv6: Disabled]
Clock Source for BPF:   ktime
Controller Status:      24/24 healthy
  Name                                  Last success   Last error     Count   Message
  cilium-health-ep                      16s ago        never          0       no error   
  dns-garbage-collector-job             27s ago        never          0       no error   
  endpoint-2534-regeneration-recovery   never          never          0       no error   
  endpoint-3923-regeneration-recovery   never          never          0       no error   
  endpoint-4044-regeneration-recovery   never          never          0       no error   
  endpoint-gc                           27s ago        never          0       no error   
  ipcache-inject-labels                 2h35m8s ago    2h35m25s ago   0       no error   
  k8s-heartbeat                         26s ago        never          0       no error   
  mark-k8s-node-as-available            2h35m17s ago   never          0       no error   
  metricsmap-bpf-prom-sync              5s ago         never          0       no error   
  neighbor-table-refresh                17s ago        never          0       no error   
  resolve-identity-2534                 16s ago        never          0       no error   
  resolve-identity-3923                 1m33s ago      never          0       no error   
  resolve-identity-4044                 17s ago        never          0       no error   
  sync-endpoints-and-host-ips           17s ago        never          0       no error   
  sync-lb-maps-with-k8s-services        2h35m17s ago   never          0       no error   
  sync-policymap-2534                   14s ago        never          0       no error   
  sync-policymap-3923                   14s ago        never          0       no error   
  sync-policymap-4044                   14s ago        never          0       no error   
  sync-to-k8s-ciliumendpoint (2534)     6s ago         never          0       no error   
  sync-to-k8s-ciliumendpoint (3923)     3s ago         never          0       no error   
  sync-to-k8s-ciliumendpoint (4044)     7s ago         never          0       no error   
  template-dir-watcher                  never          never          0       no error   
  update-k8s-node-annotations           2h35m25s ago   never          0       no error   
Proxy Status:   OK, ip 172.16.1.134, 0 redirects active on ports 10000-20000
Hubble:         Ok   Current/Max Flows: 4095/4095 (100.00%), Flows/s: 3.61   Metrics: Disabled
KubeProxyReplacement Details:
  Status:                 Strict
  Socket LB Protocols:    TCP, UDP
  Devices:                eth1 192.168.33.20 (Direct Routing)
  Mode:                   SNAT
  Backend Selection:      Random
  Session Affinity:       Enabled
  Graceful Termination:   Enabled
  XDP Acceleration:       Disabled
  Services:
  - ClusterIP:      Enabled
  - NodePort:       Enabled (Range: 30000-32767) 
  - LoadBalancer:   Enabled 
  - externalIPs:    Enabled 
  - HostPort:       Enabled
BPF Maps:   dynamic sizing: on (ratio: 0.002500)
  Name                          Size
  Non-TCP connection tracking   65536
  TCP connection tracking       131072
  Endpoint policy               65535
  Events                        2
  IP cache                      512000
  IP masquerading agent         16384
  IPv4 fragmentation            8192
  IPv4 service                  65536
  IPv6 service                  65536
  IPv4 service backend          65536
  IPv6 service backend          65536
  IPv4 service reverse NAT      65536
  IPv6 service reverse NAT      65536
  Metrics                       1024
  NAT                           131072
  Neighbor table                131072
  Global policy                 16384
  Per endpoint policy           65536
  Session affinity              65536
  Signal                        2
  Sockmap                       65535
  Sock reverse NAT              65536
  Tunnel                        65536
Encryption:           Disabled
Cluster health:       6/6 reachable   (2021-12-30T21:15:14Z)
  Name                IP              Node        Endpoints
  node2 (localhost)   192.168.33.20   reachable   reachable
  node1               192.168.33.10   reachable   reachable
  node3               192.168.33.30   reachable   reachable
  node4               192.168.33.40   reachable   reachable
  node5               192.168.33.50   reachable   reachable
  node6               192.168.33.60   reachable   reachable
```

I'm running Rocky Linux 8.5.
My Helm overrides:
```
values:
  hostPort:
    enabled: true
  hostServices:
    enabled: true
  containerRuntime:
    integration: crio
  hostFirewall:
    enabled: true
  hubble:
    relay:
      enabled: true
    ui:
      enabled: true
      ingress:
        enabled: true
        className: "{{ ingress_class }}"
        hosts: ["{{ ingress_hosts['hubble'] }}"]
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aditighag** [comment]
Can you run tcpdump with the verbose flag, and post the output?

**Comment by ooraini** [comment]
@aditighag 

```
[vagrant@node2 ~]$ sudo tcpdump -i eth1  dst 192.168.33.30 -vv
dropped privs to tcpdump
tcpdump: listening on eth1, link-type EN10MB (Ethernet), capture size 262144 bytes
01:28:55.537730 IP (tos 0x0, ttl 64, id 8646, offset 0, flags [DF], proto TCP (6), length 52)
    node2.4240 > node3.41400: Flags [.], cksum 0xc3a9 (incorrect -> 0x2eed), seq 567405791, ack 3602018924, win 227, options [nop,nop,TS val 2007678153 ecr 4204153414], length 0
01:28:55.537809 IP (tos 0x0, ttl 64, id 61203, offset 0, flags [none], proto UDP (17), length 102)
    node2.58745 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 47944, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.2.138.4240 > 172.16.3.141.46474: Flags [.], cksum 0xf79e (correct), seq 26654723, ack 3427496070, win 219, options [nop,nop,TS val 3523666377 ecr 2948071420], length 0
01:29:05.345907 IP (tos 0x0, ttl 64, id 47075, offset 0, flags [DF], proto TCP (6), length 52)
    node2.57474 > node3.4240: Flags [.], cksum 0xc3a9 (incorrect -> 0xbb31), seq 1646965779, ack 4134548851, win 229, options [nop,nop,TS val 2007687961 ecr 4204162707], length 0
01:29:05.346060 IP (tos 0x0, ttl 64, id 62371, offset 0, flags [none], proto UDP (17), length 102)
    node2.34270 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 17313, offset 0, flags [DF], proto TCP (6), length 52)
    node2.59224 > 172.16.3.47.4240: Flags [.], cksum 0x455f (correct), seq 3305426519, ack 1714877741, win 221, options [nop,nop,TS val 4195337377 ecr 1959377010], length 0
01:29:08.929814 ARP, Ethernet (len 6), IPv4 (len 4), Reply node2 is-at 52:54:00:1e:f5:6d (oui Unknown), length 28
01:29:10.466029 IP (tos 0x0, ttl 64, id 8647, offset 0, flags [DF], proto TCP (6), length 52)
    node2.4240 > node3.41400: Flags [.], cksum 0xc3a9 (incorrect -> 0xb84e), seq 1, ack 1, win 227, options [nop,nop,TS val 2007693081 ecr 4204168851], length 0
01:29:10.466109 IP (tos 0x0, ttl 64, id 62671, offset 0, flags [none], proto UDP (17), length 102)
    node2.58745 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 47945, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.2.138.4240 > 172.16.3.141.46474: Flags [.], cksum 0x8100 (correct), seq 1, ack 1, win 219, options [nop,nop,TS val 3523681305 ecr 2948086857], length 0
01:29:10.897670 IP (tos 0x0, ttl 64, id 8648, offset 0, flags [DF], proto TCP (6), length 52)
    node2.4240 > node3.41400: Flags [.], cksum 0xc3a9 (incorrect -> 0xb69f), seq 0, ack 1, win 227, options [nop,nop,TS val 2007693513 ecr 4204168851], length 0
01:29:10.897769 IP (tos 0x0, ttl 64, id 62826, offset 0, flags [none], proto UDP (17), length 102)
    node2.58745 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 47946, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.2.138.4240 > 172.16.3.141.46474: Flags [.], cksum 0x7f51 (correct), seq 0, ack 1, win 219, options [nop,nop,TS val 3523681737 ecr 2948086857], length 0
01:29:19.290936 IP (tos 0x0, ttl 64, id 53810, offset 0, flags [DF], proto ICMP (1), length 36)
    node2 > node3: ICMP echo request, id 49277, seq 33210, length 16
01:29:19.290986 IP (tos 0x0, ttl 64, id 1556, offset 0, flags [none], proto UDP (17), length 86)
    node2.49613 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 32456, offset 0, flags [DF], proto ICMP (1), length 36)
    node2 > 172.16.3.47: ICMP echo request, id 49277, seq 33210, length 16
01:29:19.292838 IP (tos 0x0, ttl 64, id 1557, offset 0, flags [none], proto UDP (17), length 204)
    node2.34270 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 17314, offset 0, flags [DF], proto TCP (6), length 154)
    node2.59224 > 172.16.3.47.4240: Flags [P.], cksum 0x682a (correct), seq 0:102, ack 1, win 221, options [nop,nop,TS val 4195351324 ecr 1959377010], length 102
01:29:19.293106 IP (tos 0x0, ttl 64, id 1558, offset 0, flags [none], proto UDP (17), length 102)
    node2.34270 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 17315, offset 0, flags [DF], proto TCP (6), length 52)
    node2.59224 > 172.16.3.47.4240: Flags [.], cksum 0x9967 (correct), seq 102, ack 76, win 221, options [nop,nop,TS val 4195351324 ecr 1959406909], length 0
01:29:19.293213 IP (tos 0x0, ttl 64, id 47076, offset 0, flags [DF], proto TCP (6), length 156)
    node2.57474 > node3.4240: Flags [P.], cksum 0xc411 (incorrect -> 0xaec4), seq 0:104, ack 1, win 229, options [nop,nop,TS val 2007701908 ecr 4204162707], length 104
01:29:19.293404 IP (tos 0x0, ttl 64, id 47077, offset 0, flags [DF], proto TCP (6), length 52)
    node2.57474 > node3.4240: Flags [.], cksum 0xc3a9 (incorrect -> 0x0f38), seq 104, ack 76, win 229, options [nop,nop,TS val 2007701908 ecr 4204192606], length 0
01:29:26.257696 IP (tos 0x0, ttl 64, id 8649, offset 0, flags [DF], proto TCP (6), length 52)
    node2.4240 > node3.41400: Flags [.], cksum 0xc3a9 (incorrect -> 0x3e9f), seq 0, ack 1, win 227, options [nop,nop,TS val 2007708873 ecr 4204184211], length 0
01:29:26.257759 IP (tos 0x0, ttl 64, id 6463, offset 0, flags [none], proto UDP (17), length 102)
    node2.58745 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 47947, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.2.138.4240 > 172.16.3.141.46474: Flags [.], cksum 0x0751 (correct), seq 0, ack 1, win 219, options [nop,nop,TS val 3523697097 ecr 2948102217], length 0
```


```
[vagrant@node2 ~]$ sudo tcpdump -i eth1  src 192.168.33.30 -vv
dropped privs to tcpdump
tcpdump: listening on eth1, link-type EN10MB (Ethernet), capture size 262144 bytes
01:30:10.801852 IP (tos 0x0, ttl 64, id 5510, offset 0, flags [DF], proto TCP (6), length 52)
    node3.41400 > node2.4240: Flags [.], cksum 0xc3a9 (incorrect -> 0xa797), seq 3602019028, ack 567405867, win 229, options [nop,nop,TS val 4204244115 ecr 2007752985], length 0
01:30:10.801923 IP (tos 0x0, ttl 64, id 36144, offset 0, flags [none], proto UDP (17), length 102)
    node3.41443 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 26788, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.3.141.46474 > 172.16.2.138.4240: Flags [.], cksum 0x704a (correct), seq 3427496173, ack 26654799, win 221, options [nop,nop,TS val 2948162121 ecr 3523741209], length 0
01:30:19.290437 IP (tos 0x0, ttl 64, id 632, offset 0, flags [none], proto ICMP (1), length 36)
    node3 > node2: ICMP echo reply, id 46548, seq 42435, length 16
01:30:19.290502 IP (tos 0x0, ttl 64, id 43024, offset 0, flags [none], proto UDP (17), length 86)
    node3.34899 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 45392, offset 0, flags [none], proto ICMP (1), length 36)
    172.16.3.47 > node2: ICMP echo reply, id 46548, seq 42435, length 16
01:30:19.291286 IP (tos 0x0, ttl 64, id 59278, offset 0, flags [DF], proto TCP (6), length 127)
    node3.4240 > node2.57474: Flags [P.], cksum 0xc3f4 (incorrect -> 0x4150), seq 4134548926:4134549001, ack 1646965987, win 227, options [nop,nop,TS val 4204252604 ecr 2007761906], length 75
01:30:19.291671 IP (tos 0x0, ttl 64, id 43025, offset 0, flags [none], proto UDP (17), length 177)
    node3.59871 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 56860, offset 0, flags [DF], proto TCP (6), length 127)
    172.16.3.47.4240 > node2.59224: Flags [P.], cksum 0xcb81 (correct), seq 1714877816:1714877891, ack 3305426723, win 219, options [nop,nop,TS val 1959466907 ecr 4195411322], length 75
01:30:26.161862 IP (tos 0x0, ttl 64, id 5511, offset 0, flags [DF], proto TCP (6), length 52)
    node3.41400 > node2.4240: Flags [.], cksum 0xc3a9 (incorrect -> 0x6b97), seq 0, ack 1, win 229, options [nop,nop,TS val 4204259475 ecr 2007752985], length 0
01:30:26.161966 IP (tos 0x0, ttl 64, id 44462, offset 0, flags [none], proto UDP (17), length 102)
    node3.41443 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 26789, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.3.141.46474 > 172.16.2.138.4240: Flags [.], cksum 0x344a (correct), seq 0, ack 1, win 221, options [nop,nop,TS val 2948177481 ecr 3523741209], length 0
01:30:34.433872 IP (tos 0x0, ttl 64, id 59279, offset 0, flags [DF], proto TCP (6), length 52)
    node3.4240 > node2.57474: Flags [.], cksum 0xc3a9 (incorrect -> 0xfea2), seq 74, ack 1, win 227, options [nop,nop,TS val 4204267747 ecr 2007761906], length 0
01:30:34.433934 IP (tos 0x0, ttl 64, id 45305, offset 0, flags [none], proto UDP (17), length 102)
    node3.59871 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 56861, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.3.47.4240 > node2.59224: Flags [.], cksum 0x88d3 (correct), seq 74, ack 1, win 219, options [nop,nop,TS val 1959482050 ecr 4195411323], length 0
01:30:40.100183 IP (tos 0x0, ttl 64, id 46494, offset 0, flags [none], proto UDP (17), length 86)
    node3.60754 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 26305, offset 0, flags [DF], proto ICMP (1), length 36)
    172.16.3.141 > 172.16.2.138: ICMP echo request, id 10964, seq 59231, length 16
01:30:40.100392 IP (tos 0x0, ttl 64, id 12212, offset 0, flags [DF], proto ICMP (1), length 36)
    node3 > node2: ICMP echo request, id 10964, seq 59231, length 16
01:30:40.101249 IP (tos 0x0, ttl 64, id 5512, offset 0, flags [DF], proto TCP (6), length 156)
    node3.41400 > node2.4240: Flags [P.], cksum 0xc411 (incorrect -> 0x6032), seq 0:104, ack 1, win 229, options [nop,nop,TS val 4204273414 ecr 2007752985], length 104
01:30:40.101524 IP (tos 0x0, ttl 64, id 5513, offset 0, flags [DF], proto TCP (6), length 52)
    node3.41400 > node2.4240: Flags [.], cksum 0xc3a9 (incorrect -> 0xc04d), seq 104, ack 76, win 229, options [nop,nop,TS val 4204273414 ecr 2007782716], length 0
01:30:40.101637 IP (tos 0x0, ttl 64, id 46495, offset 0, flags [none], proto UDP (17), length 205)
    node3.41443 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 26790, offset 0, flags [DF], proto TCP (6), length 155)
    172.16.3.141.46474 > 172.16.2.138.4240: Flags [P.], cksum 0x77cb (correct), seq 0:103, ack 1, win 221, options [nop,nop,TS val 2948191420 ecr 3523741209], length 103
01:30:40.101914 IP (tos 0x0, ttl 64, id 46496, offset 0, flags [none], proto UDP (17), length 102)
    node3.41443 > node2.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 26791, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.3.141.46474 > 172.16.2.138.4240: Flags [.], cksum 0x88ff (correct), seq 103, ack 76, win 221, options [nop,nop,TS val 2948191421 ecr 3523770941], length 0
```

**Comment by aditighag** [comment]
I only see TCP or UDP traffic in the dump. Do you identify the affected keepalived traffic (maybe using the src/dst port)? What type of L4 protocol is used by the keepalived service?

>Sysdump
https://file.io/Aai7Vju4o9I3

The file is deleted.

**Comment by ooraini** [comment]
@aditighag  The above dump was taken from a cluster running Cilium 1.11.0. There is no VRRP packets in the dump, which is precisely the issue.

The following dump is taken from cluster running Cilium 1.10.4
```
[vagrant@node2 ~]$ sudo tcpdump -i eth1  dst 192.168.33.30 -vv | grep -i vrrp
dropped privs to tcpdump
tcpdump: listening on eth1, link-type EN10MB (Ethernet), capture size 262144 bytes
23:20:21.202247 IP (tos 0xc0, ttl 255, id 4101, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:22.202302 IP (tos 0xc0, ttl 255, id 4102, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:23.202351 IP (tos 0xc0, ttl 255, id 4103, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:24.202440 IP (tos 0xc0, ttl 255, id 4104, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:25.202515 IP (tos 0xc0, ttl 255, id 4105, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:26.202568 IP (tos 0xc0, ttl 255, id 4106, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:27.202615 IP (tos 0xc0, ttl 255, id 4107, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:28.202694 IP (tos 0xc0, ttl 255, id 4108, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:29.202766 IP (tos 0xc0, ttl 255, id 4109, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 101, authtype none, intvl 1s, length 20, addrs: node2
23:20:30.202836 IP (tos 0xc0, ttl 255, id 4110, offset 0, flags [none], proto VRRP (112), length 40)
```

The issue is easily reproducible.

I will recreate the cluster with 1.11.0 and collect the sysdump again.

**Comment by aditighag** [comment]
Cilium datapath only allows packets of L4 type TCP or UDP. I see that you've enabled host firewall on cilium v1.11. Was it also enabled on 1.10 where you don't see the dropped packets?

**Comment by ooraini** [comment]
I'm deploying with same Helm values. I just change the chart version. I'll disable the host firewall on 1.11 and test.

**Comment by ooraini** [comment]
Update:
With `hostfirewall` disabled, it works on 1.11.

```
[vagrant@node1 ~]$ k exec -n kube-system cilium-5l4dg -- cilium status --verbose
Defaulted container "cilium-agent" out of: cilium-agent, clean-cilium-state (init)
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.22 (v1.22.4) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Strict   [eth1 192.168.33.30 (Direct Routing)]
Host firewall:          Disabled
Cilium:                 Ok   1.11.0 (v1.11.0-27e0848)
NodeMonitor:            Listening for events on 2 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok
IPAM:                   IPv4: 2/254 allocated from 172.16.4.0/24,
Allocated addresses:
  172.16.4.110 (health)
  172.16.4.75 (router)
BandwidthManager:       Disabled
Host Routing:           Legacy
Masquerading:           IPTables [IPv4: Enabled, IPv6: Disabled]
Clock Source for BPF:   ktime
Controller Status:      20/20 healthy
  Name                                  Last success   Last error   Count   Message
  cilium-health-ep                      40s ago        never        0       no error
  dns-garbage-collector-job             50s ago        never        0       no error
  endpoint-2455-regeneration-recovery   never          never        0       no error
  endpoint-747-regeneration-recovery    never          never        0       no error
  endpoint-gc                           50s ago        never        0       no error
  ipcache-inject-labels                 21s ago        49s ago      0       no error
  k8s-heartbeat                         20s ago        never        0       no error
  mark-k8s-node-as-available            41s ago        never        0       no error
  metricsmap-bpf-prom-sync              5s ago         never        0       no error
  neighbor-table-refresh                11s ago        never        0       no error
  resolve-identity-2455                 41s ago        never        0       no error
  resolve-identity-747                  40s ago        never        0       no error
  sync-endpoints-and-host-ips           41s ago        never        0       no error
  sync-lb-maps-with-k8s-services        41s ago        never        0       no error
  sync-policymap-2455                   21s ago        never        0       no error
  sync-policymap-747                    21s ago        never        0       no error
  sync-to-k8s-ciliumendpoint (2455)     1s ago         never        0       no error
  sync-to-k8s-ciliumendpoint (747)      10s ago        never        0       no error
  template-dir-watcher                  never          never        0       no error
  update-k8s-node-annotations           49s ago        never        0       no error
Proxy Status:   OK, ip 172.16.4.75, 0 redirects active on ports 10000-20000
Hubble:         Ok   Current/Max Flows: 113/4095 (2.76%), Flows/s: 1.38   Metrics: Ok
KubeProxyReplacement Details:
  Status:                 Strict
  Socket LB Protocols:    TCP, UDP
  Devices:                eth1 192.168.33.30 (Direct Routing)
  Mode:                   SNAT
  Backend Selection:      Random
  Session Affinity:       Enabled
  Graceful Termination:   Enabled
  XDP Acceleration:       Disabled
  Services:
  - ClusterIP:      Enabled
  - NodePort:       Enabled (Range: 30000-32767)
  - LoadBalancer:   Enabled
  - externalIPs:    Enabled
  - HostPort:       Enabled
BPF Maps:   dynamic sizing: on (ratio: 0.002500)
  Name                          Size
  Non-TCP connection tracking   65536
  TCP connection tracking       131072
  Endpoint policy               65535
  Events                        2
  IP cache                      512000
  IP masquerading agent         16384
  IPv4 fragmentation            8192
  IPv4 service                  65536
  IPv6 service                  65536
  IPv4 service backend          65536
  IPv6 service backend          65536
  IPv4 service reverse NAT      65536
  IPv6 service reverse NAT      65536
  Metrics                       1024
  NAT                           131072
  Neighbor table                131072
  Global policy                 16384
  Per endpoint policy           65536
  Session affinity              65536
  Signal                        2
  Sockmap                       65535
  Sock reverse NAT              65536
  Tunnel                        65536
Encryption:           Disabled
Cluster health:       0/2 reachable   (2022-01-05T23:27:03Z)
  Name                IP              Node        Endpoints
  node3 (localhost)   192.168.33.30   reachable   unreachable
  node4               192.168.33.40   reachable   unreachabl
```

tcpdump:
```
[vagrant@node2 ~]$ sudo tcpdump -i eth1  dst 192.168.33.30 -vv
dropped privs to tcpdump
tcpdump: listening on eth1, link-type EN10MB (Ethernet), capture size 262144 bytes
02:28:18.063851 IP (tos 0xc0, ttl 255, id 15, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:19.063929 IP (tos 0xc0, ttl 255, id 16, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:20.064058 IP (tos 0xc0, ttl 255, id 17, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:21.064117 IP (tos 0xc0, ttl 255, id 18, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:21.735867 IP (tos 0x0, ttl 64, id 41066, offset 0, flags [DF], proto ICMP (1), length 36)
    node2 > node3: ICMP echo request, id 5801, seq 28884, length 16
02:28:21.735975 IP (tos 0x0, ttl 64, id 28510, offset 0, flags [none], proto UDP (17), length 86)
    node2.58002 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 42396, offset 0, flags [DF], proto ICMP (1), length 36)
    node2 > 172.16.4.110: ICMP echo request, id 5801, seq 28884, length 16
02:28:21.752250 IP (tos 0x0, ttl 64, id 8598, offset 0, flags [DF], proto TCP (6), length 156)
    node2.47048 > node3.4240: Flags [P.], cksum 0xc411 (incorrect -> 0xdf29), seq 4187871681:4187871785, ack 1387051343, win 229, options [nop,nop,TS val 1701205895 ecr 2146703523], length 104
02:28:21.756363 IP (tos 0x0, ttl 64, id 8599, offset 0, flags [DF], proto TCP (6), length 52)
    node2.47048 > node3.4240: Flags [.], cksum 0xc3a9 (incorrect -> 0x4027), seq 104, ack 76, win 229, options [nop,nop,TS val 1701205899 ecr 2146733280], length 0
02:28:21.756478 IP (tos 0x0, ttl 64, id 28516, offset 0, flags [none], proto UDP (17), length 205)
    node2.38988 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 8850, offset 0, flags [DF], proto TCP (6), length 155)
    node2.56944 > 172.16.4.110.4240: Flags [P.], cksum 0x35fa (correct), seq 2508166341:2508166444, ack 3330621361, win 221, options [nop,nop,TS val 2033113901 ecr 4010671922], length 103
02:28:21.764420 IP (tos 0x0, ttl 64, id 28519, offset 0, flags [none], proto UDP (17), length 102)
    node2.38988 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 6
IP (tos 0x0, ttl 64, id 8851, offset 0, flags [DF], proto TCP (6), length 52)
    node2.56944 > 172.16.4.110.4240: Flags [.], cksum 0x4104 (correct), seq 103, ack 76, win 221, options [nop,nop,TS val 2033113909 ecr 4010701687], length 0
02:28:22.064874 IP (tos 0xc0, ttl 255, id 19, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:23.064927 IP (tos 0xc0, ttl 255, id 20, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:24.064983 IP (tos 0xc0, ttl 255, id 21, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:25.065103 IP (tos 0xc0, ttl 255, id 22, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:25.791841 IP (tos 0x0, ttl 64, id 44077, offset 0, flags [DF], proto TCP (6), length 52)
    node2.4240 > node3.53774: Flags [.], cksum 0xc3a9 (incorrect -> 0x801b), seq 2371349721, ack 812840731, win 227, options [nop,nop,TS val 1701209935 ecr 2146722099], length 0
02:28:25.792230 IP (tos 0x0, ttl 64, id 29960, offset 0, flags [none], proto UDP (17), length 102)
    node2.51551 > node3.otv: [no cksum] OTV, flags [I] (0x08), overlay 0, instance 4
IP (tos 0x0, ttl 64, id 45185, offset 0, flags [DF], proto TCP (6), length 52)
    172.16.1.230.4240 > 172.16.4.75.33956: Flags [.], cksum 0x02a8 (correct), seq 1096957319, ack 548196857, win 219, options [nop,nop,TS val 2630268084 ecr 3594446442], length 0
02:28:26.065177 IP (tos 0xc0, ttl 255, id 23, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:27.065259 IP (tos 0xc0, ttl 255, id 24, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
02:28:28.065325 IP (tos 0xc0, ttl 255, id 25, offset 0, flags [none], proto VRRP (112), length 40)
    node2 > node3: vrrp node2 > node3: VRRPv2, Advertisement, vrid 51, prio 99, authtype none, intvl 1s, length 20, addrs: node2
```

**Comment by ooraini** [comment]
Keepalived is running on the host with the following configuration:
```
global_defs {
    router_id LVS_DEVEL
    enable_script_security
}
vrrp_script check {
  script "/usr/libexec/keepalived/check"
  interval 3
  weight -2
  fall 5
  rise 2
  init_fail
}

vrrp_instance VI_1 {
    state MASTER
    interface eth1
    virtual_router_id 51
    priority 101

    check_unicast_src

    unicast_peer {
        192.168.33.30
    }

    virtual_ipaddress {
        192.168.33.200
    }

    track_script {
        check
    }
}
```

**Comment by aditighag** [comment]
>Update:
With hostfirewall disabled, it works on 1.11.

Thanks for following up. 

What's the output of this command when run from one of the cilium pods - `sudo bpftool prog show | grep from_netdev`?

If you need to enable host firewall in your cluster without affecting keepalived traffic, you could potentially have keepalived use a separate node interface. Exclude that interface from the `--devices` config passed to cilium so that BPF programs that drop non-supported L4 protocols won't be involved. Cc @pchaigno to confirm this.

**Comment by ooraini** [comment]
> 



> > Update:
> > With hostfirewall disabled, it works on 1.11.
> 
> Thanks for following up.
> 
> What's the output of this command when run from one of the cilium pods - `sudo bpftool prog show | grep from_netdev`?
> 
On 1.11 with `hostfirewall` disabled:
```
Defaulted container "cilium-agent" out of: cilium-agent, clean-cilium-state (init)
765: sched_cls  name from_netdev  tag 043688a319dd8e5e  gpl
```
> If you need to enable host firewall in your cluster without affecting keepalived traffic, you could potentially have keepalived use a separate node interface. Exclude that interface from the `--devices` config passed to cilium so that BPF programs that drop non-supported L4 protocols won't be involved. Cc @pchaigno to confirm this.

Note that it did work with `hostfirewall` _enabled_ in 1.10. But not in 1.11.

Thanks for the help.

**Comment by pchaigno** [comment]
The behavior change is unexpected AFAIK. I also can't think of a change in v1.11 that could explain this. Next step would probably be to bisect.

**Comment by aditighag** [comment]
>Note that it did work with hostfirewall enabled in 1.10. But not in 1.11.

The 1.11 behavior seems expected since the datapath drops packets with L4 protocol other than TCP, UDP, or ICMP. One thing to check between 1.10 and 1.11 clusters is whether the bpf_host program is attached on the interface that's used for the keep alive traffic.

You can run `sudo bpftool net show` on the cilium pod in both 1.10 and 1.11, with host-fw enabled.

**Comment by ooraini** [comment]
1.10.4:
```
❯ k exec cilium-2xt2q -- bpftool net show
Defaulted container "cilium-agent" out of: cilium-agent, clean-cilium-state (init)
xdp:

tc:
eth1(3) clsact/ingress bpf_netdev_eth1.o:[from-netdev] id 701
eth1(3) clsact/egress bpf_netdev_eth1.o:[to-netdev] id 707
cilium_net(4) clsact/ingress bpf_host_cilium_net.o:[to-host] id 695
cilium_host(5) clsact/ingress bpf_host.o:[to-host] id 683
cilium_host(5) clsact/egress bpf_host.o:[from-host] id 689
cilium_vxlan(6) clsact/ingress bpf_overlay.o:[from-overlay] id 645
cilium_vxlan(6) clsact/egress bpf_overlay.o:[to-overlay] id 650
lxc_health(8) clsact/ingress bpf_lxc.o:[from-container] id 667
lxc024fb7ec7e53(10) clsact/ingress bpf_lxc.o:[from-container] id 665
lxcbbcd04f5cea6(12) clsact/ingress bpf_lxc.o:[from-container] id 669

flow_dissector:



```

1.11:
```
Defaulted container "cilium-agent" out of: cilium-agent, clean-cilium-state (init)
xdp:

tc:
eth1(3) clsact/ingress bpf_netdev_eth1.o:[from-netdev] id 769
eth1(3) clsact/egress bpf_netdev_eth1.o:[to-netdev] id 779
cilium_net(4) clsact/ingress bpf_host_cilium_net.o:[to-host] id 759
cilium_host(5) clsact/ingress bpf_host.o:[to-host] id 739
cilium_host(5) clsact/egress bpf_host.o:[from-host] id 749
cilium_vxlan(6) clsact/ingress bpf_overlay.o:[from-overlay] id 670
cilium_vxlan(6) clsact/egress bpf_overlay.o:[to-overlay] id 678
lxc_health(8) clsact/ingress bpf_lxc.o:[from-container] id 727
lxca71197083523(10) clsact/ingress bpf_lxc.o:[from-container] id 788

flow_dissector:


```

**Comment by clrxbl** [comment]
This still appears to be an issue and breaks VRRP usage on Cilium 1.12 host firewall enabled machines.

**Comment by jm-bertheas-datailor** [comment]
Same problem on Clilum 1.13.2

**Comment by sepich** [comment]
A possible workaround for `keepalived` could be `MetalLB arp mode`
https://metallb.universe.tf/concepts/layer2/
because it uses in-cluster `memberlist` instead of `vrrp` for healthchecks

**Comment by akurz** [comment]
We are using `purelb` with instances of `bird` doing OSPF with our Cisco SDN to exchange dynamic route informations. Using cilium 1.13.4 I see the egress multicast packages being dropped due to `CT: Unknown L4 protocol` onc the host-firewall is enabled. Would be great to whitelist protocol names/numbers to be able to enable the host-firewall.

**Comment by lgyurci** [comment]
OSPF not working is my problem as well, and I see the exact same behavior described by @akurz

**Comment by taraszka** [comment]
It's the same with Cilium 1.15.1, I am afraid. The workaround with excluding that interface from the `--devices` config passed to cilium isn't the best; why? Well... usually, it is a public interface that you want to protect with Cilium's host firewall rules (instead of using a Linux firewall (firewalld, ufw, etc.)

**Comment by kvaster** [comment]
It seems that keepalived is broken even without host firewall enabled on 1.15.1.

I've applied following patch in order to fix this for me (at least for now):

```
--- a/bpf/bpf_host.c
+++ b/bpf/bpf_host.c
@@ -1321,9 +1321,9 @@ int cil_to_netdev(struct __ctx_buff *ctx __maybe_unused)
        __u32 __maybe_unused vlan_id;
        int ret = CTX_ACT_OK;
        __s8 ext_err = 0;
-#ifdef ENABLE_HOST_FIREWALL
+//#ifdef ENABLE_HOST_FIREWALL
        __u16 proto = 0;
-#endif
+//#endif
 
        /* Filter allowed vlan id's and pass them back to kernel.
         */
@@ -1338,6 +1338,19 @@ int cil_to_netdev(struct __ctx_buff *ctx __maybe_unused)
                }
        }
 
+       if (validate_ethertype(ctx, &proto)) {
+               void *data, *data_end;
+               struct iphdr __maybe_unused *ip4;
+               switch (proto) {
+                       case bpf_htons(ETH_P_IP):
+                               if (revalidate_data(ctx, &data, &data_end, &ip4)) {
+                                       if (ip4->daddr >= bpf_htonl(0xE0000000) && ip4->daddr <= bpf_htonl(0xEFFFFFFF)) {
+                                               return CTX_ACT_OK;
+                                       }
+                       }
+               }
+       }
+
 #if defined(ENABLE_L7_LB)
        {
                __u32 magic = ctx->mark & MARK_MAGIC_HOST_MASK;
```

**Comment by sbocahu** [comment]
Can this issue be retitled to address the current issues that are targeting 1.15 (and not only and old version, with the risk for the issue to stale); that do not require hostFirewall to be enabled and impact broader protocols/traffic than VRRP ?

**Comment by KarstenSiemer** [comment]
Hi everyone! Is there any progress on somehow supporting other protocols next to TCP/UDP like VRRP or CARP?
Or has someone found a workaround to have VRRP traffic with hostPolicies activated?

Any help would be greatly appreciated!

**Comment by minhng99** [comment]
Hi, I am trying to use Cilium for an on-prem setup with the control-planes running VRRP and having the same issue

**Comment by kvaster** [comment]
For now you must have host firewall turned off. Or you need to allow such traffic in an explicit way.

**Comment by larssb** [comment]
Wow! Recently became aware of this issue. We also use Keepalived for on-prem VMWare running K3s clusters control-plane. Disabling `hostFirewall` at the `Cilium` level makes `VRRP` traffic come through.

This is on `Cilium v1.15.5`. Pretty wild that this has been an issue since late 2021 and we still don't have a proper fix.

I guess one can, to an extend, live without using `Host Policies`. However, some `Cilium Network Policies` will be much more effective if implemented via at the host level.

Cool [workaround](https://github.com/cilium/cilium/issues/18347#issuecomment-1998334558) @kvaster - However, I don't feel confident in this approach in our case. At least not until we're more well wandered within the ecosystem of `eBPF`.

Any `Cilium org.` members with a hot take on this one?

Thank you very much.

**Comment by minhng99** [comment]
I've found that if you keep Cilium running, it will block the multicast traffic between the control planes which makes keepalived always answer its ARP on every control plane nodes, this behavior is unintended but it does have a nice side effects which allows the control plane to be "load balanced" because the workers will connect to whichever nodes send its arp first. 

I know this isn't how it supposed to work... But so far i have zero issue with it, I suggest you can try leave it in

**Comment by larssb** [comment]
Hmm @minhng99 thank you for responding. Interesting take indeed. It's not our experience that `keepalived` works reliably enough in the case that `Cilium HostFirewall` is being used.

But, again. Thank you very much.

**Comment by lgyurci** [comment]
In the comments we have also mentioned the need for ospf, but I dont see that covered by this pull request. Should I open a separate issue for it? We would still need this feature
